lib/bolt/secret.rb

```diff
@@ -1,17 +1,37 @@
 # frozen_string_literal: true
 
+require 'bolt/plugin'
+
 module Bolt
   class Secret
+    KNOWN_KEYS = {
+      'createkeys' => %w[keysize private_key public_key],
+      'encrypt'    => %w[public_key],
+      'decrypt'    => %w[private_key public_key]
+    }.freeze
+
     def self.execute(plugins, outputter, options)
-      plugin = options[:plugin] || 'pkcs7'
+      name   = options[:plugin] || 'pkcs7'
+      plugin = plugins.by_name(name)
+
+      unless plugin
+        raise Bolt::Plugin::PluginError::Unknown, name
+      end
+
+      opts = plugin.config.slice(*KNOWN_KEYS[options[:action]])
```

Comment:
I'm not sure I follow this. Shouldn't the plugin system already deal with pulling out the desired plugin config? It seems like it's `options` we ought to be pulling keys out of (and renaming) here.

Reply:
Okay yeah it looks like this causes the CLI flags for `--private-key`, etc to be ignored.